webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_never_granted_bar.tsx

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import AnnouncementBar from 'components/announcement_bar/default_announcement_bar';
+
+import {AnnouncementBarTypes} from 'utils/constants';
+import {requestNotificationPermission} from 'utils/notifications';
+
+export default function NotificationPermissionNeverGrantedBar() {
+    const [show, setShow] = useState(true);
+
+    const handleClick = useCallback(async () => {
+        await requestNotificationPermission();
+
+        // Dismiss the bar after user makes a choice
+        setShow(false);
+    }, []);
+
+    const handleClose = useCallback(() => {
+        // If the user closes the bar, don't show the notification bar any more for the rest of the session, but
+        // show it again on app refresh.
+        setShow(false);
+    }, []);
+
+    if (!show) {
+        return null;
+    }
+
+    return (
+        <AnnouncementBar
+            showCloseButton={true}
+            handleClose={handleClose}
+            type={AnnouncementBarTypes.ANNOUNCEMENT}
+            message={
+                <FormattedMessage
+                    id='announcementBar.notification.permissionNeverGrantedBar.message'
+                    defaultMessage='We need your permission to show web browser notifications.'
```

Comment:
Well I think `web browser notifications` technically makes sense but I also agree that almost all other products call them `desktop notifications`. I wouldn't want to change it to just "We need your permission to show notifications" since it can leave room for ambiguity and could be confused with in-product mention badges and what not. 

I'm open to reverting back to `desktop notifications` but I'd want @cwarnermm to share thoughts and take the final call on this since this change was recommended by her.

Reply:
How about "We need your permission to show notifications in this web browser" ?

Reply:
I don't mind it. I was thinking it's getting a bit long but we have the space and it doesn't truncate until the screen size is less than ~805px so that's okay.

<img width="1527" alt="image" src="https://github.com/user-attachments/assets/b8212009-f959-42c4-9b7d-d66c865c5962">

Thoughts @hmhealey?

Reply:
How about "We need your permission to show notifications in the browser"? I still have a vendetta against calling it a "web browser" apparently, and while "this" technically correct that it only refers to the one browser, I feel like that doesn't matter to most people since they just use a single one

Reply:
changed as discussed above